SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -363,6 +366,143 @@ void InterestFactory::AddServerGameplayDebuggerCategoryReplicatorActorInterest(I
 }
 #endif
 
+bool UnrealServerInterestFactory::CreateClientInterestDiff(const AActor* InActor, Worker_CommandRequest& Request,
+														   const bool bOverwrite) const
+{
+	const APlayerController* PlayerController = Cast<APlayerController>(InActor);
+	if (!ensure(GetDefault<USpatialGDKSettings>()->bUseClientEntityInterestQueries)
+		|| !ensureMsgf(PlayerController != nullptr, TEXT("Only PCs can use client entity interest: %s"), *GetNameSafe(InActor)))
+	{
+		return false;
+	}
+
+	// If and when the Runtime has a better interest changing API, we can ask for a diff here instead.
+	TArray<Worker_EntityId> ClientInterestedEntities = GetClientInterestedEntityIds(PlayerController);
+
+	USpatialNetDriver* NetDriver = Cast<USpatialNetDriver>(PlayerController->GetNetDriver());
+
+	// MAKE DEBUGGING EASIER, REVERT ME
+#if WITH_EDITOR
+	ClientInterestedEntities.Sort();
+#endif
+
+	static ChangeInterestRequest ChangeInterestRequestData;
```

Comment:
Sanity check, it's only global statics that mess things up with multiple servers in a single process?

Reply:
Yep - this here is very much a hack to prevent the ChangeInterestRequest data going out of scope when returned from the function - so it can be passed into the SpatialConnection. Really I should just move this to a member variable, or make the caller pass it in.